docs/src/pages/system/basics/basics.md

```diff
@@ -186,6 +186,14 @@ Cons:
 
   We believe that for most uses it's **fast enough**, but there are simple workarounds when performance becomes critical. For instance, when rendering a list with many items, you can use a CSS child selector to have a single "style injection" point (using d. for the wrapper and a. for each item).
 
+### API tradeoff
+
+In the previous versions, we supported the system properties as first class properties on the `Box` component. However, as we are extending the system to be a superset of CSS (contains all CSS properties/selectors in addition to custom ones) it's no longer an option to support all properties as first class props. This is the main reason why, we decided to support all system properties under one prop - `sx`, which in addition allows developers to define other CSS definitions.
```

Comment:
```suggestion
In the previous versions, we supported the system properties as props on the `Box` component. However, the system is a superset of CSS and selectors cannot be efficiently mapped to props without potential conflicts. This is the main reason why, we decided to support all system properties under one prop - `sx`, which in addition allows developers to define other CSS definitions.
```

1. I don't think there's a notion of "first-class props", just props
1. As far as I can tell it's not about mapping system properties to props but rather that we can't efficiently map selectors. So we consolidate it under a single prop to avoid refactoring hazard when going from "just system properties" to system properties + selectors. Kind of like how inline styles should be avoided because you'll eventually have to refactor to style rules.

Reply:
Agreed, and even more, is not just about the selectors, the there are many other CSS Properties, that are not part of the system, which we would need to map to props...

Reply:
I still think we should describe what superset of CSS means, as it seems like it is confusing for people. What do you think of this - https://github.com/mui-org/material-ui/pull/23962/files#r541019182 as final?

Reply:
I've applied https://github.com/mui-org/material-ui/pull/23962/files#r541019182 , feel free to leave suggestion if anything else needs to be updated